backend/src/sockets/disconnect.js

```diff
@@ -1,23 +1,38 @@
-import games from '../domain/Games';
+import games from "../domain/Games";
 /**
  * Socket event for when a user disconnects.
  * @param {*} socket
  * @param {*} io
  * @param {*} cb callback function
  */
 export function disconnect(socket, io, cb) {
-  socket.on('disconnect', () => {
+  socket.on("disconnect", () => {
     const activeGames = games.getGames();
 
     activeGames.forEach((game) => {
+      const users = Array.from(game.session.users.values()).map(
+        (e) => e.name
+      );
+      // Indicate host has left
+      if (game.session.hostSocket && socket.id === game.session.hostSocket.id) {
+        game.session.hostSocket = null;
+      }
+      // Remove users with same socket
       if (game.session.removeUser(socket)) {
-        const users = Array.from(
-            game.session.users.values()).map((e) => e.name);
+        // Recalculate users as have been changed
+        users = Array.from(game.session.users.values()).map((e) => e.name);
         console.log(users);
-        io.to(game.session.sessionId).emit('new_user', {users: users});
+        io.to(game.session.sessionId).emit("new_user", { users: users });
```

Comment:
Another nitpick but not totally necessary: could extract this common functionality out